frontend/src/metabase/ui/components/inputs/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,61 @@
+import { useLayoutEffect, useState } from "react";
+import type { ChangeEvent } from "react";
+import moment from "moment-timezone";
+import { TimeInput as MantineTimeInput } from "@mantine/dates";
+import type { TimeInputProps as MantineTimeInputProps } from "@mantine/dates";
+
+const TIME_FORMAT = "HH:mm";
+
+export type TimeInputProps = Omit<
+  MantineTimeInputProps,
+  "value" | "defaultValue" | "onChange"
+> & {
+  value?: Date | null;
+  defaultValue?: Date | null;
+  onChange?: (value: Date) => void;
+};
+
+export function TimeInput({
+  value,
+  defaultValue = value,
+  onChange,
+  ...props
+}: TimeInputProps) {
```

Comment:
Curios why we need `value` and `defaultValue`? Can't we do something like that:

```jsx
function TimeInput({ value, onChange }) {
  const inputValue = value ? formatTime(value) : ""
  
  const handleChange = event => {
    const newTime = parseTime(event.target.value)
    onChange(newTime)
  }

  return <MantineTimeInput value={inputValue} onChange={handleChange} />
}
```

Reply:
`defalueValue` is needed for uncontrolled usage (all mantine components support it)

Reply:
e.g. `<TimeInput defaultValue={new Date()} />` - no `onChange` handler

Reply:
Do we need uncontrolled inputs though?

Reply:
all current mantine inputs support uncontrolled usage

Reply:
Oh, I see, makes sense to keep things consistent